signer/src/blocklist_client.rs

```diff
@@ -0,0 +1,169 @@
+//! # Blocklist Client Module
+//!
+//! This module provides the `BlocklistChecker` trait and its `BlocklistClient` implementation,
+//! which are used to check addresses against a blocklist service. The module's responsibilities
+//! include querying the blocklist API and interpreting the responses to determine if a given
+//! address is blocklisted, along with its associated risk severity.
+
+use crate::config::SETTINGS;
+use async_trait::async_trait;
+use blocklist_client::common::error::Error;
+use blocklist_client::common::BlocklistStatus;
```

Comment:
Hmmm, I don't think we should import the blocklist client. Instead, we should have the blocklist client have an API and we build the types from the API. One way is to have an OpenAPI spec that we build the types from. Another approach is to write protobufs and build the types that way.

This is not a simple change, so what are your thoughts on this?

Reply:
I am open to those more robust approaches...Since the client is small and only exposes one API, I thought this type of dependency/API invocation would be sufficient.
@netrome  any preferences here?

Reply:
Generating an API and building types from it seems like an excessive amount of work compared to just importing the types directly, leveraging the fact that we share a common code base.

In this scenario I'm more in favor of keeping things lean from now, although I would agree with the approach for a more extensive API with multiple consumers.